src/coprocessor/dag/expr/builtin_time.rs

```diff
@@ -148,17 +172,75 @@ impl ScalarFunc {
 
 #[cfg(test)]
 mod tests {
-    use coprocessor::codec::mysql::Time;
+    use coprocessor::codec::mysql::{Duration, Time};
     use coprocessor::codec::Datum;
     use coprocessor::dag::expr::tests::{datum_expr, scalar_func_expr};
     use coprocessor::dag::expr::*;
     use coprocessor::dag::expr::{EvalContext, Expression};
     use std::sync::Arc;
     use tipb::expression::ScalarFuncSig;
 
+    fn test_ok_case_one_arg(ctx: &mut EvalContext, sig: ScalarFuncSig, arg: Datum, exp: Datum) {
+        let arg = datum_expr(arg);
+        let f = scalar_func_expr(sig, &[arg]);
+        let op = Expression::build(ctx, f).unwrap();
+        let got = op.eval(ctx, &[]);
+        match got {
+            Ok(got) => assert_eq!(got, exp),
+            Err(_) => assert!(false, "eval failed"),
+        }
+    }
+
+    fn test_err_case_one_arg(ctx: &mut EvalContext, sig: ScalarFuncSig, arg: Datum, exp: Datum) {
+        let arg = datum_expr(arg);
+        let f = scalar_func_expr(sig, &[arg]);
+        let op = Expression::build(ctx, f).unwrap();
+        let got = op.eval(ctx, &[]);
+        match got {
+            Ok(got) => assert_eq!(got, exp),
+            Err(_) => assert!(true),
+        }
+    }
+
+    fn test_ok_case_two_arg(
```

Comment:
Missing `#[test]`

Reply:
Functions used as tests can not have any arguments.
These are just functions to help tests.

Reply:
Oh, sorry.